junit-runtime/src/main/scala/scala/scalanative/junit/Bootstrapper.scala

```diff
@@ -0,0 +1,40 @@
+package scala.scalanative
+package junit
+
+// Ported from Scala.js
+
+import scala.scalanative.reflect.annotation._
+import scala.util.Try
+
+/** Scala Native internal JUnit bootstrapper.
+ *
+ *  This class is public due to implementation details. Only the junit compiler
+ *  plugin may generate classes inheriting from it.
+ *
+ *  Relying on this trait directly is unspecified behavior.
+ */
+@EnableReflectiveInstantiation
+trait Bootstrapper {
+  def beforeClass(): Unit
+  def afterClass(): Unit
+  def before(instance: AnyRef): Unit
+  def after(instance: AnyRef): Unit
+
+  def tests(): Array[TestMetadata]
+  def invokeTest(instance: AnyRef, name: String): Try[Unit]
```

Comment:
Even if you dropped support for async tests to make things work for now, we must prepare the interface of `Bootstrapper` for that support later on. Therefore, `invokeTest` should still return `Future[Try[Unit]]` and the JUnit plugin should still wrap results in `Future.successful`.